pkg/create/sample-cluster.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package create
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+)
+
+var githubURL = "https://raw.githubusercontent.com/rook/rook/master/deploy/examples/"
+
+func SampleCluster(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace, clusterNamespace, flag string) {
+	var answer, output string
+	logging.Warning(`
+	"This command will create a single-node Rook-Ceph cluster. We assume that you don't already have a Rook-Ceph cluster.
+	We also assume that you have a Kubernetes cluster and have read this https://rook.github.io/docs/rook/latest/Contributing/development-environment/ . Please type 'yes' to proceed."
+	`)
+
+	fmt.Scanf("%s", &answer)
+	output, err := printWarningMessage(answer)
+	if err != nil {
+		logging.Fatal(fmt.Errorf("Single node Rook-Ceph cluster creation aborted: %w", err))
+	}
+	logging.Info(output)
+
+	crdsYaml, err := getFileContentFromGitHub(githubURL, "crds.yaml", operatorNamespace, clusterNamespace)
+	if err != nil {
+		logging.Error(fmt.Errorf("Error fetching file crds.yaml: %v", err))
+		return
+	}
+	createK8sResources("crds.yaml", crdsYaml)
+
+	commonYaml, err := getFileContentFromGitHub(githubURL, "common.yaml", operatorNamespace, clusterNamespace)
+	if err != nil {
+		logging.Error(fmt.Errorf("Error fetching file common.yaml: %v", err))
+		return
+	}
+	createK8sResources("crds.yaml", commonYaml)
+
+	operatorYaml, err := getFileContentFromGitHub(githubURL, "operator.yaml", operatorNamespace, clusterNamespace)
+	if err != nil {
+		logging.Error(fmt.Errorf("Error fetching file operator.yaml: %v", err))
+		return
+	}
+	createK8sResources("crds.yaml", operatorYaml)
+
+	clusterYaml, err := getFileContentFromGitHub(githubURL, "cluster-test.yaml", operatorNamespace, clusterNamespace)
```

Comment:
I anticipate many users will have trouble to get this to work when they don't realize they need devices available. It may also surprise them that this test will consume all devices with this cluster-test.yaml. Perhaps we should only provide this command for the cluster-on-pvc case. Then we don't have to worry if there are devices on the nodes. It's commonly more simple to cleanup the cluster as well, since it's just a test cluster. But let's discuss...